src/core/server/connector/pipe_connector_posix.h

```diff
@@ -0,0 +1,34 @@
+#ifndef CORE_SERVER_CONNECTOR_PIPE_CONNECTOR_POSIX_H_
+#define CORE_SERVER_CONNECTOR_PIPE_CONNECTOR_POSIX_H_
+
+#include <cstdio>
+#include <string>
+
+#include "core/server/connector/pipe_connector.h"
+
+struct FrameRequest;
+struct FrameResponse;
+
+class PipeConnectorPosix : public PipeConnector {
+public:
+    static std::unique_ptr<Connector> create(int playerId, const std::string& program);
+
+    virtual ~PipeConnectorPosix();
+
+    virtual bool waitAndReceive(bool waitTimeout, int frameId, std::vector<FrameResponse>* cfr) final;
+
+private:
+    PipeConnectorPosix(int player, int writerFd, int readerFd);
+
+    virtual void writeString(const std::string&) final;
+    virtual bool readString(char*) final;
+
+    FILE* writer_;
+    FILE* reader_;
+
+    int writerFd_;
+    int readerFd_;
+};
+
+#endif
```

Comment:
Could you make this like this?
#endif // CORE_SERVER_CONNECTOR_PIPE_CONNECTOR_POSIX_H_